### PR TITLE
Improving structure and error handling

### PR DESCRIPTION
**Pull request by leovms**
- Normalized task scripts to make their structure more consistent.
- Improved error handling to recognize cases where bad data is passed.
- Updating encoding from UTF16 to UTF8.

### CODE REVIEW

File: VstsTasks/AzureDtlCreateVM/task-funcs.ps1 [hunk 1]
```diff
@@ -0,0 +1,180 @@
+function Handle-LastError
```
Comment (joerage) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: You have duplication of some methods, anyway to have a shared func file between the folders?

File: VstsTasks/AzureDtlCreateVM/task-funcs.ps1 [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+function Get-TemplateParameterValue
+{
+    [CmdletBinding()]
+    Param(
+        [string]$Parameters,
+        [string]$ParameterName
+    )
+
+    $pattern = '\-(?<k>\w+)\s+(?<v>\''.*?\''|\$\(.*\)?|\(.*\)?)'
```
Comment (joerage) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: Can you add a comment with an example match to help undersand the code?

File: VstsTasks/AzureDtlCreateVM/task-funcs.ps1 [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        $mustReplaceDefaults = $true
+    }
+
+    if ($mustReplaceDefaults)
+    {
+        throw 'Default values must be replaced. Please review the Template Parameters and modify as needed.'
+    }
+}
+
+function Validate-VMName
```
Comment (joerage) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: We don't check for reserved name. Worth doing it here?

File: VstsTasks/AzureDtlDeleteVM/task-funcs.ps1 [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    [CmdletBinding()]
+    param(
+    )
+
+    Write-Host "Task called with the following parameters:"
+    Write-Host "  ConnectedServiceName = $ConnectedServiceName"
+    Write-Host "  LabVMId = $LabVMId"
+}
+
+function Invoke-AzureDtlTask
```
Comment (joerage) [inline VstsTasks/AzureDtlDeleteVM/task-funcs.ps1]: Why a generic name? Why not Invoke-DeleteDtlLab

File: VstsTasks/AzureDtlCreateVM/task-funcs.ps1 (hunk 1 above)
Comment (leovms) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: Unfortunately, I cannot do anything about that as the task expects all its files to be relative, within the folder structure.

File: VstsTasks/AzureDtlCreateVM/task-funcs.ps1 (hunk 3 above)
Comment (leovms) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: We let the RP return with the error, so as to minimize duplicate checks that may change over time and would therefore make it difficult to maintain this code.

File: VstsTasks/AzureDtlDeleteVM/task-funcs.ps1 (hunk 4 above)
Comment (leovms) [inline VstsTasks/AzureDtlDeleteVM/task-funcs.ps1]: To make the intended structure clear.

File: VstsTasks/AzureDtlCreateVM/task-funcs.ps1 (hunk 2 above)
Comment (leovms) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: Do you want a comment on the regex? Is that what you ask?

Comment (joerage) [inline VstsTasks/AzureDtlCreateVM/task-funcs.ps1]: Yep!

**Pull request merged**